src/test/sql/sql_translator_test.cpp

```diff
@@ -1080,6 +1095,23 @@ TEST_F(SQLTranslatorTest, FromColumnAliasingSimple) {
   EXPECT_LQP_EQ(actual_lqp_b, expected_lqp);
 }
 
+TEST_F(SQLTranslatorTest, FromColumnAliasingColumnsSwitchNames) {
+  // Tricky: Columns "switch names". a becomes b and b becomes a
+
+  const auto actual_lqp_a = compile_query("SELECT * FROM int_float AS t (b, a) WHERE b = t.a");
+  const auto actual_lqp_b = compile_query("SELECT * FROM (SELECT * FROM int_float) AS t (b, a) WHERE b = t.a");
+
+  // clang-format off
+  const auto expected_lqp =
+  AliasNode::make(expression_vector(int_float_a, int_float_b), std::vector<std::string>({"b", "a"}),
+    PredicateNode::make(equals_(int_float_a, int_float_b),
+      stored_table_node_int_float));
+  // clang-format on
+
+  EXPECT_LQP_EQ(actual_lqp_a, expected_lqp);
+  EXPECT_LQP_EQ(actual_lqp_b, expected_lqp);
+}
+
```

Comment:
(random line)
`SELECT a AS b, b AS a FROM t WHERE a > 5` should now yield an InputError, right? If not, add tests for it. If so, add it to `TEST_F(SQLTranslatorTest, CatchInputErrors)`

Reply:
Yes, I added a test for the case that this query yields an 'InvalidInputException'.